src/scenario/mod.rs

```diff
@@ -248,6 +253,20 @@ limit 1
             Ok::<String, anyhow::Error>(purl.to_string())
         })
     }
+
+    /// A purl ID for details lookup
+    pub async fn purl_details(&self) -> anyhow::Result<String> {
```

Comment:
Looks good. But I think the result should be a bit more predictable, or focused. Like finding a PURL with "the most advisories referenced", or whatever makes the request costly. Similar to the other SQLs.

Reply:
Yes. That's the idea. I think using a more complex query makes sense.